R/tm_g_spiderplot.R

```diff
@@ -236,14 +239,27 @@ ui_g_spider <- function(id, ...) {
   )
 }
 
-srv_g_spider <- function(id, data, filter_panel_api, reporter, dataname, label, plot_height, plot_width) {
+srv_g_spider <- function(id, data, filter_panel_api, paramcd, reporter, dataname, label, plot_height, plot_width) {
   with_reporter <- !missing(reporter) && inherits(reporter, "Reporter")
   with_filter <- !missing(filter_panel_api) && inherits(filter_panel_api, "FilterPanelAPI")
   checkmate::assert_class(data, "reactive")
   checkmate::assert_class(shiny::isolate(data()), "teal_data")
 
   moduleServer(id, function(input, output, session) {
     teal.logger::log_shiny_input_changes(input, namespace = "teal.osprey")
+
+    isolate({
+      env <- as.list(data()@env)
+      resolved_paramcd <- teal.transform::resolve_delayed(paramcd, env)
+
+      teal.widgets::updateOptionalSelectInput(
+        session = session,
+        inputId = "paramcd",
+        choices = resolved_paramcd$choices,
+        selected = resolved_paramcd$selected
+      )
+    })
```

Comment:
Can we add explicit `isolate` to only places where this is intended for?
Because, this will make it clear what we're trying to isolate. In this case it looks like we need to stop reactivity on `data()` so it would be nice if we could just call a `isolate(data())` instead.

Reply:
Please make sure to isolate in other places affected in this PR too :)

Reply:
Good catch, Vedha. I have updated it as per your suggestion.
Commit: 4cb9f0b